### DISCUSSION TITLE
我发现长文本生成效果不容易调出来，特别容易出现重复 例如“我吃饭了吗吗吗吗吗吗吗吗吗”

### DISCUSSION THREAD

**Discussion by liuanping** [zai-org/ChatGLM-6B#1378] [Q&A]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

我发现长文本生成效果不容易调出来，特别容易出现重复 例如“我吃饭了吗吗吗吗吗吗吗吗吗”，网上说是退化问题，即随着生成文本长度的增加其质量会逐渐降低，容易出现多种层次（字、短语、句子级）的重复生成。有没有大神给一些有效的经验。

### Expected Behavior

求大神指点

### Steps To Reproduce

训练长文本生成。

### Environment

```markdown
- OS:ubuntu-18
- Python:3.8.13
- Transformers:4.28
- PyTorch:2.2
- CUDA Support (`python -c "import torch; print(torch.cuda.is_available())"`) :11.7
```

### Anything else?

none

**Comment by StarRanger** [comment]
你用的p-tuning吗？还是lora？

**Comment by liuanping** [comment]
> 你用的p-tuning吗？还是lora？
用的是lora

**Comment by Crazycatter** [comment]
我也是这个问题 减少文本长度有用么？

**Comment by xv994** [comment]
我在alpaca-lora遇到了同样的问题，猜测生成长文本的难度确实较大，而且lora在微调方法中也算是属于效果不太好的那一类了

**Comment by Leawnn** [comment]
我用p-tuning也是出现这种
求大神解答

**Comment by StarRanger** [comment]
max_target_length别指定太长，缩短到64，训练次数适当增加。部署时，tockenizer就用原本的Thudm/ChatGLM-6B，model用ptuning的model，你这情况我在训练llama时遇到过。chatglm没遇见。

发自我的iPhone

**Comment by Leawnn** [comment]
> 

训练的时候max_target_length设置的300，max_steps设置5000，还是不太行
![image](https://user-images.githubusercontent.com/56593035/233292101-236f9363-d0d5-46c8-98ac-b9b4b1e6513a.png)

**Comment by Leawnn** [comment]
![image](https://user-images.githubusercontent.com/56593035/233294856-e60cfe8e-c077-4057-81c2-8130144fbebe.png)

**Comment by xv994** [comment]
> ![image](https://user-images.githubusercontent.com/56593035/233294856-e60cfe8e-c077-4057-81c2-8130144fbebe.png)

max_target_length为128时，生成中文文本字数大概为200字左右，你这个max_target_length=300，我猜测如果基底模型生成的内容不够长的话，就会不停重复

**Comment by SnakeHacker** [comment]
同样遇到了，有时候会疯狂排比句

**Comment by liuanping** [comment]
用全量微调 感觉没有这种问题了

**Comment by Lufffya** [comment]
@liuanping 大佬解决了吗，我也出现了同样的问题

**Comment by liuanping** [comment]
> 
感觉全量微调就好了

**Comment by liuanping** [comment]
@Lufffya 感觉全量微调就好了 还有一些办法说是加重复惩罚项

**Comment by Lufffya** [comment]
> @Lufffya 感觉全量微调就好了 还有一些办法说是加重复惩罚项

哦哦好吧，那估计不太行，我这里只有一张4090，跑不起来，谢谢

**Comment by liuanping** [comment]
@Lufffya 可能chatuan更友好  因为他是10亿参数模型 效果也还行。

**Comment by Chevalier1024** [comment]
> @Lufffya 感觉全量微调就好了 还有一些办法说是加重复惩罚项

请问你有全量微调嘛

**Comment by shuanglong520** [comment]
@liuanping 大佬解决了吗，我也出现了同样的问题
有没有怎么微调相关的说明，谢谢

**Comment by liuanping** [comment]
@shuanglong520 我全量微调的时候发现没了 lora不行可能
